### ISSUE TITLE
PhET-iO design for Basics version

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/geometric-optics-basics#3]
@arouinfar and I discussed the general requirement in https://github.com/phetsims/geometric-optics-basics/issues/2, and how what they mean for PhET-iO instrumentation. 

Changing defaults, and showing/hiding things in the sim is easy. On the other end of the spectrum is PhET-iO: it's difficult to change what PhET-iO elements are created, so that things that are not relevant to Basics are never created, and don't appear in the Studio tree. We're not sure how to proceed, and should have a general discussion.

In the meantime, I'm going to take the absolute easiest approach to make the sim look & behave as desired.

**Comment by pixelzoom** [comment]
@samreid commented in #2, which is about general requirements. I'm moving that comment to this issue, which is about PhET-iO design.  Responding to the comment above, @samreid said:

> On the other end of the spectrum is PhET-iO: it's difficult to change what PhET-iO elements are created, so that things that are not relevant to Basics are never created

We have faced this in Circuit Construction Kit and in Center and Spread.  For instance, in center and spread, this code only instruments the velocity of an object in the soccer screens:

```ts
this.velocityProperty = new Vector2Property( options.velocity, {
  tandem: objectType === CASObjectType.SOCCER_BALL ?
          options.tandem.createTandem( 'velocityProperty' ) :
          Tandem.OPT_OUT
} );
```

To clarify.  The `velocityProperty` is always created, but a PhET-iO element is only created when appropriate.

We have also adjusted tandem names as appropriate for the context:

```ts
this.addLinkedElement( casObject, {
  tandem: options.tandem.createTandem(
    casObject.objectType === CASObjectType.SOCCER_BALL ? 'soccerBall' :
    'dataPoint'
  ),
  phetioState: false
} );
```

**Comment by pixelzoom** [comment]
Yes thanks, I'm familiar with how it can be accomplished, and I've done conditional instrumentation in the past.  It's more difficult to implement, more difficult to test fully, and harder to maintain.  So the question is: Does PhET want to incur the cost to do this? 

If PhET does decide to go down the path of conditionally instrumenting based on Basics vs Full versions, here are the Pros and Cons:

**Pros**
* The API is arguably simpler, since the Basics version won't contain as many things that are not relevant. 

**Cons**
* More difficult/costly to design, implement, test, maintain.
* Instructional Designers will need to learn how to navigate 2 different Studio trees.
* Client code will be more difficult to repurpose/tweak because the APIs will be significantly different.

**Pro or Con?**
* The client will not have the option of restoring something that PhETs has decided to remove from the Basics version. For example, if the client prefers to use a concave lens in their Basics version, that will be impossible.

**Comment by pixelzoom** [comment]
Here's an idea.. Instead of a costly development effort to create a variation of a sim, why doesn't PhET use the PhET-iO API to create a version like Basics? Right now, I can take the full version of Geometric Optics, and customize it in Studio to look exactly like what we want for the Basics version.  But I can't turn that into something that PhET can publish.  Is using PhET-iO to create and publish variations of PhET sims something that should be pursued?  Creating something like Geometric Optics: Basics could conceivably be done by a design in a few hours, rather than a developer in weeks or months.

**Comment by samreid** [comment]
I proposed a similar idea for simplifying the creation of sim variations in https://github.com/phetsims/chipper/issues/1153#issuecomment-1035373219.  Many of those pros and cons also apply to this proposal.

**Comment by pixelzoom** [comment]
Update: _Geometric Optics: Basics_ was feature-complete < 24 hours after the creation of its repository.  The PhET-iO API (and Studio tree) is currently identical to _Geometric Optics_, and I consider that to be advantageous for PhET and clients.  If PhET decided to conditionally uninstrument/rename PhET-iO elements for Basics, I suspect that's going to take many days (if not weeks) of design and implementation.

**Comment by arouinfar** [comment]
@pixelzoom did you forget to push to master?  I've pulled all and cleared the cache, but the Studio tree for geometric-optics-basics is empty except for `phetioEngine`.

**Comment by samreid** [comment]
This may be due to a phet-io problem.  I'm seeing that the `simStarted` event is not processing properly, so there is an error in the console.  Probably not a geometric-optics-specific problem.  Tagging @zepumph in case he has ideas.

**Comment by arouinfar** [comment]
Thanks @samreid looks like it is indeed a general problem. I tried a few other sims and they're all showing the same empty tree.

**Comment by samreid** [comment]
Thanks, I moved it to a side issue.

**Comment by samreid** [comment]
OK I pushed a fix in the issue.  @arouinfar can you please pull and test?

**Comment by pixelzoom** [comment]
This Studio problem is unrelated to this issue. So in an effort to keep this issue clean, I'm moving those comments to https://github.com/phetsims/geometric-optics-basics/issues/4.

**Comment by pixelzoom** [comment]
2/24/2022 design meeting:

The consensus was that "one API, one Studio tree" is a good fit for these sims.   

I'm going to restore the various optics shapes, so that they can be restored via Studio, if desired.

I also notices that there's a layout problem when 'direct' is the default. The UI components to the left of the control panel are in a bad location.

**Comment by pixelzoom** [comment]
Change requests from 2/24/2022 design meeting were completed in the above commits. 

Ready for PhET-iO design and review.
